### PR TITLE
Add a script to patch in task deadlines where required

### PR DESCRIPTION
**Pull request by wheelsandcogs**

### CODE REVIEW

**Comment by lennym** [comment]
Could you wrap it all in a transaction please?

File: bin/add-deadlines-to-legacy-tasks [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+};
+
+const migrate = async () => {
+  const db = Database.connect(settings.taskflowDB);
+  Task.db(db);
+
+  // only need to patch tasks that have already been submitted, because the deadline object gets added on submission
+  const withAsruOrComplete = withASRU().concat(['resolved', 'rejected']);
+
+  const query = Task.query()
```
Comment (lennym) [inline bin/add-deadlines-to-legacy-tasks:61]:
```suggestion
  const tasks = await Task.query()
```

File: bin/add-deadlines-to-legacy-tasks [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    .select('*')
+    .whereIn('status', withAsruOrComplete)
+    .whereJsonSupersetOf('data', { model: 'project', action: 'grant' })
+    .whereRaw("data->>'deadline' is null"); // will match both deadline: null and missing deadline key
+
+  const tasks = await query;
```
Comment (lennym) [inline bin/add-deadlines-to-legacy-tasks:67]:
```suggestion
```

File: bin/add-deadlines-to-legacy-tasks (hunk 1 above)
Comment (wheelsandcogs) [reply bin/add-deadlines-to-legacy-tasks:61]: This was because I was debugging the query sql 👍

File: bin/add-deadlines-to-legacy-tasks [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    .whereJsonSupersetOf('data', { model: 'project', action: 'grant' })
+    .whereRaw("data->>'deadline' is null"); // will match both deadline: null and missing deadline key
+
+  console.log(`found ${tasks.length} tasks with no deadline property set`);
+
+  for (const task of tasks) {
+    await patchTaskDeadline(task.id, transaction); // process the tasks one at a time to avoid ballooning memory
+  }
+
+  await transaction.commit();
```
Comment (lennym) [inline bin/add-deadlines-to-legacy-tasks:75]: The transaction will need to be rolled back in case of an error.

**Pull request merged**